esmvaltool/cmor/_fixes/CMIP5/MRI_CGCM3.py

```diff
@@ -0,0 +1,28 @@
+"""Fixes for MRI-CGCM3 model"""
+import numpy as np
+
+from ..fix import Fix
+
+
+class msftmyz(Fix):
+    """Fixes for msftmyz."""
+    def fix_data(self, cube):
+        """
+        Fix data
+
+        Fixes mask
+
+        Parameters
+        ----------
+        cube: iris.cube.Cube
+
+        Returns
+        -------
+        iris.cube.Cube
+
+        """
+        cube.data = np.ma.array(cube.data)
+        cube.data = np.ma.masked_where(cube.data.mask + (cube.data==0.),
```

Comment:
Lee, what are you actually doing here?

Reply:
it'd be nicer if you worked with masks independently of the data objects eg:
```
zero_mask = cube.data==0.
if cube.data.mask:
    cumm_mask = zero_mask & cube.data.mask (can use np.ma.mask_or(zero_mask, native_mask))
    cube.data = np.ma.array(cube.data, mask=cumm_mask)
else:
    cube.data = np.ma.array(cube.data, mask=zero_mask)
```

This way you are avoiding masking ahead for an unmasked cube in memory

Reply:
also, is there a scientific reason why these mask files are masked this way ie are you 100% sure they're broekn? I wouldn't fix them everywhere if the actual native (and seemingly broken mask) will actually be needed for some other diagnostics

Reply:
You can also add a fix for the specific variable you are using.

Reply:
With regards to the CCSM4 model type problem, the problem appears to me to be that the regions that should be masked mistakenly take the values of the closest unmasked data of the same depth. 

This is especially visible on the left hand side, but also can be seen in the bottom right corner of this figure, where the water column should be much shallower. 

![diag_cmip5_ccsm4_omon_historical_r1i1p1_to2m_msftmyz_prep_transect_diag_transects_2000_2004_transect](https://user-images.githubusercontent.com/4180685/49743698-3c7cbd00-fc93-11e8-9dca-efe27793ccad.png)

As a reminder, this is a zonal mean of an irregular region (the Atlantic ocean), and it's non-trivial to produce a mask for this data.

It's not clear to me how to fix this. We may have to live with it.